.travis.yml

```diff
@@ -35,6 +38,29 @@ after_success:
   - pip install codecov
   - codecov
 
+matrix:
+  allow_failures:
+    - os: osx
+  include:
+    - name: "osx + builtin python3"
+      os: osx
+      language: sh
+      before_cache:
+              - rm -f "$HOME/Library/Caches/pip/log/debug.log"
+      cache:
+        directories:
+          - "$HOME/Library/Caches/pip"
+      before_install:
```

Comment:
What's different to the other jobs? Why do we need to define this step again. The individual steps look pretty familiar

Reply:
The cache location is different on osx and simply setting `cache: pip` didn't seem to work as expected for me (I did some searching, I think this is due to Travis not officially supporting python on osx).

Reply:
I was referring to everything else than the cache

Reply:
Right. So: 

a) `before_install` doesn't stack
b) we need to explicitly setup a virtual environment in case of osx (which happens by default for linux), so the common steps need essentially to be redefined.

Reply:
And splitting it out into a common shell script doesn't work as expected. For example: https://travis-ci.com/kagharpure/kartothek/jobs/219631966

Reply:
Actually, hang on; I think sourcing the script ought to fix this because it's then not running it in a separate shell - just checking now.

Reply:
The linked build fails because the requirements pinning wasn't successful. not because the build script was wrong.

I'm wondering if the setup of the virtualenv can't be used for all builds. I don't see a reason why mac is special here. The same script should run on linux as well

Reply:
> The linked build fails because the requirements pinning wasn't successful. not because the build script was wrong.

That's because it's using the system `pip`, not the pip updated in the venv created above.

 > 'm wondering if the setup of the virtualenv can't be used for all builds. I don't see a reason why mac is special here. The same script should run on linux as well

If the os is linux, travis creates a venv by default, the user needn't explicitly specify that it be created.